### ISSUE TITLE
Updates fails on Nominatim 3.3.0

### ISSUE THREAD

**Issue by uilll** [openstreetmap/Nominatim#1426]
Hello everyone, I installed Nomination 3.3.0, I imported the data from Geofabric, all right according to the tutorials "http://nominatim.org/release-docs/develop/appendix/Install-on-Ubuntu-18/" and "http://nominatim.org/release-docs/develop/admin/Import-and-Update/", but at the time of updating these errors appeared:

Command 1: 

> `./utils/update.php --init-updates`

Answer:
```
2019-07-19 00:50:27 == module path: / srv / nominatim / Nominatim / build / module
2019-07-19 00:50:27 == Create Functions
BUG: HTTP Error 404: Not Found
Done. Database updates will start at sequence 3586635 (2019-07-17T17: 06: 14Z)
``` 

I believe this 404 error was not to pop because I left the default setting using the global minutely diffs.

Command 2: 

> `./utils/update.php --import-osmosis-all`

Answer:
```
Currently at sequence 3586635 (17/07/2019 20:06:14) - 1 indexed
/srv/nominatim/.local/bin/pyosmium-get-changes --server https://planet.openstreetmap.org/replication/minute -o /srv/nominatim/Nominatim/build/osmosischange.osc -s 30 -I 3586635
Traceback (most recent call last):
  File "/srv/nominatim/.local/bin/pyosmium-get-changes", line 243, in <module>
    exit (main (sys.argv [1:]))
  File "/srv/nominatim/.local/bin/pyosmium-get-changes", line 224, in main
    outhandler = WriteHandler (options.outfile)
RuntimeError: Open failed for '/srv/nominatim/Nominatim/build/osmosischange.osc': Permission denied
```

So, When checking the permissions of the osmosischange.osc file I saw that the file does not exist in this directory:

Command 

> /srv/nominatim/Nominatim/build/ ls -la

Answer:
```
total 128
drwxr-xr-x 11 root root 4096 Jul 18 14:13.
drwxr-xr-x 21 root root 4096 Jul 18 14:12 ..
-rw -r-r-- 1 root root 27560 Jul 18 14:13 CMakeCache.txt
drwxr-xr-x 4 root root 4096 Jul 18 14:15 CMakeFiles
-rw-r-r-- 1 root root 1065 Jul 18 14:13 CTestTestfile.cmake
-rw-r-r-- 1 root root 2716 Jul 18 14:13 DartConfiguration.tcl
-rw-r-r-- 1 root root 37832 Jul 18 14:13 Makefile
drwxr-xr-x 3 root root 4096 Jul 18 14:13 Testing
-rw-r-r-- 1 root root 1885 Jul 18 14:13 cmake_install.cmake
drwxr-xr-x 5 root root 4096 Jul 18 14:13 docs
-rw-r-r-- 1 root root 1575 Jul 18 14:13 mkdocs.yml
drwxr-xr-x 3 root root 4096 Jul 18 14:15 module
drwxr-xr-x 3 root root 4096 Jul 18 14:15 nominatim
drwxr-xr-x 5 root root 4096 Jul 18 14:15 osm2pgsql
drwxr-xr-x 2 root root 4096 Jul 19 00:50 settings
drwxr-xr-x 2 root root 4096 Jul 18 14:13 utils
drwxr-xr-x 2 root root 4096 Jul 18 14:13 website
```

So could anyone help me solve or better inform me about these two problems?

**Comment by mtmail** [comment]
Which user is executing the update command? I see the build directory and all files in it are owned by user `root`. The permission problem could simply be that the current user (`nominatim`?) cannot write to the directory.

The docs you linked to use user `nominatim`. It's easy to recreate the directory if needed. A recursive `chown` and `chgrp` might be easier.

```
mv build build.backup
mkdir build
cd build
cmake .. && make
cd -
cp build.backup/settings/local.php build/settings/
```

**Comment by uilll** [comment]
OK, I changed the owner and group owners, but another error appeared.

Command:
`nominatim@\my-ip:~/Nominatim/build$ ./utils/update.php --import-osmosis-all`

Response:

```
WARNING: resetting cache memory to 814
Currently at sequence 3586635 (17/07/2019 20:06:14) - 1 indexed
/srv/nominatim/.local/bin/pyosmium-get-changes --server https://planet.openstreetmap.org/replication/minute -o /srv/nominatim/Nominatim/build/osmosischange.osc -s 30 -I 3586635
sh: 1: /srv/nominatim/Nominatim/utils/osm_file_date.py: not found
ERROR: Error getting date from diff file.
string(34) "Error getting date from diff file."
```

however the file exist:

```
nominatim@My-ip:~/Nominatim/utils$ ls -la
total 108
drwxr-xr-x  2 nominatim nominatim  4096 Jul 18 14:11 .
drwxr-xr-x 21 nominatim nominatim  4096 Jul 18 14:12 ..
-rwxr-xr-x  1 nominatim nominatim   624 Jul 18 14:11 check_server_for_updates.py
-rw-r--r--  1 nominatim nominatim  1159 Jul 18 14:11 country_languages.php
-rw-r--r--  1 nominatim nominatim  8160 Jul 18 14:11 export.php
-rw-r--r--  1 nominatim nominatim 25881 Jul 18 14:11 importWikipedia.php
-rwxr-xr-x  1 nominatim nominatim  3871 Jul 18 14:11 import_wikipedia.sh
-rwxr-xr-x  1 nominatim nominatim   568 Jul 18 14:11 osm_file_date.py
-rw-r--r--  1 nominatim nominatim  2057 Jul 18 14:11 query.php
-rwxr-xr-x  1 nominatim nominatim  2062 Jul 18 14:11 server_compare.php
-rw-r--r--  1 nominatim nominatim  6424 Jul 18 14:11 setup.php
-rw-r--r--  1 nominatim nominatim  5088 Jul 18 14:11 specialphrases.php
-rw-r--r--  1 nominatim nominatim 19823 Jul 18 14:11 update.php
-rw-r--r--  1 nominatim nominatim  2283 Jul 18 14:11 warm.php
```

**Comment by mtmail** [comment]
Can you doublecheck the permissions of the parent directories? Maybe `/srv/nominatim/Nominatim/` is owned by root? Is `~` the same as `/srv/nominatim/`? I'm guessing here.

**Comment by uilll** [comment]
the "/ srv" directory and its subdirectories have the "nominatim" owner and the "nominatim" group:

```
nominatim @ my-ip: / $ ls -la
total 88
drwxr-xr-x 23 root root 4096 Jul 19 01:17.
drwxr-xr-x 23 root root 4096 Jul 19 01:17 ..
drwxr-xr-x 2 root root 4096 Jul 18 14:00 bin
drwxr-xr-x 3 root root 4096 Jun 27 16:38 boot
drwxr-xr-x 15 root root 2980 Jul 19 01:17 dev
drwxr-xr-x 99 root root 4096 Jul 19 06:34 etc
drwxr-xr-x 3 root root 4096 Jul 18 13:39 home
lrwxrwxrwx 1 root root 31 Jun 27 16:38 initrd.img -> boot / initrd.img-4.15.0-1043-aws
lrwxrwxrwx 1 root root 31 Jun 27 16:38 initrd.img.old -> boot / initrd.img-4.15.0-1043-aws
drwxr-xr-x 20 root root 4096 Jul 18 14:01 lib
drwxr-xr-x 2 root root 4096 Jun 27 16:25 lib64
drwx ------ 2 root root 16384 Jun 27 16:27 lost + found
drwxr-xr-x 2 root root 4096 Jun 27 16:25 media
drwxr-xr-x 2 root root 4096 Jun 27 16:25 mnt
drwxr-xr-x 2 root root 4096 Jun 27 16:25 opt
dr-xr-xr-x 129 root root 0 Jul 19 01:17 proc
drwx ------ 5 root root 4096 Jul 18 23:45 root
drwxr-xr-x 26 root root 920 Jul 19 13:37 run
drwxr-xr-x 2 root root 4096 Jun 27 16:27 sbin
drwxr-xr-x 5 root root 4096 Jul 18 13:39 snap

> drwxr-xr-x 3 nominatim nominatim 4096 Jul 18 14:03 srv

dr-xr-xr-x 13 root root 0 Jul 19 01:17 sys
drwxrwxrwt 10 root root 4096 Jul 19 16:09 tmp
drwxr-xr-x 10 root root 4096 Jun 27 16:25 usr
drwxr-xr-x 14 root root 4096 Jul 18 14:00 var
lrwxrwxrwx 1 root root 28 Jun 27 16:38 vmlinuz -> boot / vmlinuz-4.15.0-1043-aws
lrwxrwxrwx 1 root root 28 Jun 27 16:38 vmlinuz.old -> boot / vmlinuz-4.15.0-1043-aws
```

```
nominatim@My-ip:/srv$ ls -la
total 12
drwxr-xr-x  3 nominatim nominatim 4096 Jul 18 14:03 .
drwxr-xr-x 23 root      root      4096 Jul 19 01:17 ..
drwxr-xr-x  6 nominatim nominatim 4096 Jul 19 00:13 nominatim
```

```
nominatim@My-ip:~$ ls -la
total 44
drwxr-xr-x  6 nominatim nominatim 4096 Jul 19 00:13 .
drwxr-xr-x  3 nominatim nominatim 4096 Jul 18 14:03 ..
-rw-------  1 nominatim nominatim 7235 Jul 19 16:07 .bash_history
-rw-r--r--  1 nominatim nominatim  220 Apr  4  2018 .bash_logout
-rw-r--r--  1 nominatim nominatim 3771 Apr  4  2018 .bashrc
drwx------  3 nominatim nominatim 4096 Jul 19 00:13 .cache
drwx------  5 nominatim nominatim 4096 Jul 19 00:18 .local
-rw-r--r--  1 nominatim nominatim  807 Apr  4  2018 .profile
drwxr-xr-x 21 nominatim nominatim 4096 Jul 18 14:12 Nominatim
drwxr-xr-x  2 nominatim nominatim 4096 Jul 18 14:26 to
```

By following the same subdirectories up to `/srv/nominatim/Nominatim/utils/osm_file_date.py`, all with nominatim owner and nominatim group

**Comment by mtmail** [comment]
That looks all good.

Does executing `/srv/nominatim/Nominatim/utils/osm_file_date.py /srv/nominatim/Nominatim/build/osmosischange.osc` work? It should show a date, e.g. `2019-07-18 23:59:01`

**Comment by uilll** [comment]
Yes it's work.

```
nominatim@My-ip:~/Nominatim/build$ ls -la
total 66556
drwxr-xr-x 11 nominatim nominatim     4096 Jul 19 16:24 .
drwxr-xr-x 21 nominatim nominatim     4096 Jul 18 14:12 ..
-rw-r--r--  1 nominatim nominatim    27560 Jul 18 14:13 CMakeCache.txt
drwxr-xr-x  4 nominatim nominatim     4096 Jul 18 14:15 CMakeFiles
-rw-r--r--  1 nominatim nominatim     1065 Jul 18 14:13 CTestTestfile.cmake
-rw-r--r--  1 nominatim nominatim     2716 Jul 18 14:13 DartConfiguration.tcl
-rw-r--r--  1 nominatim nominatim    37832 Jul 18 14:13 Makefile
drwxr-xr-x  3 nominatim nominatim     4096 Jul 18 14:13 Testing
-rw-r--r--  1 nominatim nominatim     1885 Jul 18 14:13 cmake_install.cmake
drwxr-xr-x  5 nominatim nominatim     4096 Jul 18 14:13 docs
-rw-r--r--  1 nominatim nominatim     1575 Jul 18 14:13 mkdocs.yml
drwxr-xr-x  3 nominatim nominatim     4096 Jul 18 14:15 module
drwxr-xr-x  3 nominatim nominatim     4096 Jul 18 14:15 nominatim
drwxr-xr-x  5 nominatim nominatim     4096 Jul 18 14:15 osm2pgsql

> -rw-rw-r--  1 nominatim nominatim 68020965 Jul 19 16:26 osmosischange.osc

drwxr-xr-x  2 nominatim nominatim     4096 Jul 19 00:50 settings
drwxr-xr-x  2 nominatim nominatim     4096 Jul 18 14:13 utils
drwxr-xr-x  2 nominatim nominatim     4096 Jul 18 14:13 website
```

**Comment by mtmail** [comment]
So far the only difference we found is part of the installation was run as root. (There's already a bold "Never, ever run the installation as a root user." in the [installation instructions ](http://nominatim.org/release-docs/develop/appendix/Install-on-Ubuntu-18/). Closing here because we looked at all file permissions, the script itself executes fine and the underlying issue is likely not the Nominatim software. A fresh installation should work.
